board/gpio.md

```diff
@@ -3,70 +3,97 @@ GPIO API
 
 The GPIO (General Purpose Input & Output) API supports digital pins.
 
-The API object
---------------
-GPIO pin functionality is exposed by the [`GPIO`](#gpio) object that can be obtained by using the [gpio() method of the `Board` API](./README.md/#gpio). See also the [Web IDL](./webidl.md).
-
 On certain boards GPIO pins may be grouped into ports (e.g. 8, 16 or 32 pins), read and written as registers by the controller.
 
 On certain boards, analog pins can also be used as GPIO.
 
-<a name="gpio"></a>
-### The `GPIO` interface
-Represents the properties and methods that expose GPIO functionality. The `GPIO` object implements the [`EventEmitter`](../README.md/#events) interface, and extends the [`Pin`](./README.md/#pin) object.
+<a name="apiobject"></a>
+### The GPIO API object
+When requiring `"gpio"`, the following steps are run:
+- If there is no permission for using the functionality, throw `SecurityError`.
+- If the AIO functionality is not supported on the board, throw `"NotSupportedError"`.
+- Return an object that implements the following methods.
+
+| Method              | Description      |
+| ---                 | ---              |
+| [`open()`](#open)   | open GPIO pin    |
+| [`port()`](#port)   | open GPIO port   |
+
+See also the [Web IDL](./webidl.md) definition.
+
+The following dictionary is used for initializing GPIO pins and ports.
+<a name="gpiooptions"></a>
 
 | Property   | Type   | Optional | Default value | Represents |
 | ---        | ---    | ---      | ---           | ---        |
-| `pin`      | String or Number | no | `undefined`   | board name for the pin |
+| `pin `     | String or Number | no | `undefined`   | pin name |
+| `mapping`  | String | no | `"system"`   | pin mapping |
```

Comment:
I don't mean that this spec would specify those formats, I'm just saying this spec could require that if an implementation allows a different (e.g. OS-specific) format of pin naming, they do it in a way that is distinguishable from the system-specific mapping, so that a "mapping" doesn't need to be specified.